### PR TITLE
v1.9 (focused backport) backports 2021-06-28

### PR DESCRIPTION
**Pull request by aditighag**
* #16259 -- kpr: Enable/fix Cilium socket based load balancing in different environments (@aditighag)
* #16815 -- Fixes for platforms where init container fails to run -- Fixes for platforms where init container fails to run (@aditighag)

Once this PR is merged, you can update the PR labels via:
```upstream-prs
$ for pr in 16259 16815; do contrib/backporting/set-labels.py $pr done 1.9; done
```

### CODE REVIEW

File: Documentation/gettingstarted/kubeproxy-free.rst [hunk 1]
```diff
@@ -236,16 +251,16 @@ port ``31940`` (one for each of devices ``eth0`` and ``eth1``):
 .. parsed-literal::
 
     kubectl exec -it -n kube-system cilium-fmh8d -- cilium service list
-    ID   Frontend               Service Type   Backend
```
Comment (aditighag) [inline Documentation/gettingstarted/kubeproxy-free.rst:239]: I have no idea why these changes are showing up. When I do `git annotate`, I don't see them.

Comment (aanm) [reply Documentation/gettingstarted/kubeproxy-free.rst:239]: I think it's because your editor is removing the trailing spaces. It's fine.

**Comment by aditighag** [comment]
test-backport-1.9

**Comment by aditighag** [comment]
Needs specific fixes for v1.9 backport since the `experimental/install.yaml` file was removed in later releases.

>please run 'make -C install/kubernetes experimental-install' and submit your changes
Error: Process completed with exit code 1.

**Comment by aditighag** [comment]
test-backport-1.9

**Comment by joestringer** [comment]
test-backport-1.9

**Comment by aditighag** [comment]
> test-backport-1.9

@joestringer Why did you have to restart the tests?

**Comment by aditighag** [comment]
runtime failure - https://github.com/cilium/cilium/issues/16713
5.4 - known complexity issue
